ckanext/canada/cli.py

```diff
@@ -209,17 +214,34 @@ def append_log(finished, package_id, action, reason):
                 stats = completion_stats(self.processes)
                 while result is not None:
                     try:
-                        package_id, action, reason = json.loads(result)
+                        package_id, action, reason, error, failure_reason, failure_trace, do_update_sync_success_time = json.loads(result)
                     except Exception as e:
                         if self.verbose:
                             print("Worker proccess failed on:")
                             print(result)
                         raise Exception(e)
-                    print(job_ids, next(stats), finished, package_id, \
-                        action, reason)
-                    append_log(finished, package_id, action, reason)
+                    _stats = next(stats)
+                    print(job_ids, _stats, finished, package_id, action, reason)
+                    if error:
+                        print(job_ids, _stats, finished, package_id, error)
+                    append_log(finished, package_id, action, reason, error)
                     job_ids, finished, result = next(pool)
 
+                    #TODO: need to actually output error to stderr!!!
```

Comment:
it's ok for the parent to print to stdout for normal status messages, for errors you can `print(..., file=sys.stderr)`